### DISCUSSION TITLE
Will the controller spawn pod base on architecture label

### DISCUSSION THREAD

**Discussion by H01001000** [actions/actions-runner-controller#891] [Questions] [answered]
In my action, there are jobs that only run on arm64 or x64, if I use workflow_job event webhook, will the controller be able to add node selector on the pod, and therefore spawn runner with architecture that I specify on runs-on? Or do I need to make runner deployment for each architecture only?

**Comment by toast-gear** [comment, marked as answer]
> Or do I need to make runner deployment for each architecture only?

this

**Reply by H01001000** [reply]
for now what I did is having three deployments label x64-runner, arm64-runner, and a cluster runner.
